services/digest_io.py

```
@@ -1,251 +1,370 @@
+            # Load normalized digest
+            digest = self.load_normalized_digest(target_date)
             
-            logger.info(f"Created FINAL digest: {final_path}")
-            return enhanced_digest
+            # Enhance with AI
+            enriched_digest = self.enhance_with_ai(digest)
             
+            # Save enriched digest
+            self.save_enriched_digest(enriched_digest, target_date)
+            logger.info(f"Created enriched digest for {target_date}")
+            return enriched_digest
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)